### PR TITLE
updated to use localStorage, removed page state logic.

### PR DESCRIPTION
**Pull request by adphillips2017**

### CODE REVIEW

**Review by bherr2 — CHANGES_REQUESTED** [review]
Move the localStorage logic into the PageState.. This is obviously state and is most appropriate there.

File: website/src/app/core/state/page/page.state.ts [hunk 1]
```diff
@@ -19,8 +21,16 @@ interface PageStateModel {
 export class PageState extends NgxsImmutableDataRepository<PageStateModel> {
   readonly hasShownHelpModal$ = this.state$.pipe(pluck('hasShownHelpModal'));
 
+  ngxsOnInit(): void {
+    super.ngxsOnInit();
+    this.patchState({
+      hasShownHelpModal: localStorage.getItem(LOCAL_STORAGE_HELP_MODAL_KEY)?.toLowerCase() === 'true' ? true : false
```
Comment (bherr2) [inline website/src/app/core/state/page/page.state.ts:27]: This should probably move to defaults in the @State decorator (line 17) and could be simplified to just `localStorage.getItem(LOCAL_STORAGE_HELP_MODAL_KEY)?.toLowerCase() === 'true'`

**Review by bherr2 — CHANGES_REQUESTED** [review]

File: website/src/app/core/state/page/page.state.ts (hunk 1 above)
Comment (axdanbol) [reply website/src/app/core/state/page/page.state.ts:27]: I disagree. I think defaults in the @State decorator should be kept as constants and not execute a bunch of code. Also if we ever were to change from where the value is loaded it is less likely to cause problems.

Comment (bherr2) [reply website/src/app/core/state/page/page.state.ts:27]: Fair point, what about as a constant above in the file. I don't like the patchState piece.

Comment (axdanbol) [reply website/src/app/core/state/page/page.state.ts:27]: `localStorage.getItem(...)` is dynamic and should preferably be executed at the time the hasShownHelpModal value is set. There is some cleanup that can happen though. Ex:
```
ngxsOnInit(): void {
  super.ngxsOnInit();
  const hasShownHelpModal = localStorage.getItem(LOCAL_STORAGE_HELP_MODAL_KEY)?.toLowerCase() === 'true'; // Ternary operator not needed
  this.patchState({ hasShownHelpModal });
}
```

Comment (axdanbol) [reply website/src/app/core/state/page/page.state.ts:27]: Note: Using `ngxsOnInit` and `patchState` is the correct way to load and set dynamic values during state initialization

Comment (bherr2) [reply website/src/app/core/state/page/page.state.ts:27]: Ok

Comment (axdanbol) [reply website/src/app/core/state/page/page.state.ts:27]: @adphillips2017 See above for changes

**Review by bherr2 — APPROVED** [review]

**Pull request merged**